pkg/tap/tap.go

```diff
@@ -16,6 +17,11 @@ import (
 	log "github.com/sirupsen/logrus"
 )
 
+var (
+	// ErrClosedResponseBody is returned when response body is closed in http2
+	ErrClosedResponseBody = errors.New("http2: response body closed")
```

Comment:
It looks like we check for this error: `errors.Is(err, tap.ErrClosedResponseBody)` but we never actually return this error.  I notice this looks the same as the error from http2/transport:

https://go.googlesource.com/net/+/master/http2/transport.go#2158

But I don't think creating our own error with the same text will let us catch that one.  Unfortunately, that error is private and doesn't seem to implement any useful public interfaces, so this may be a case where string matching on the error message is the only way.

Reply:
@adleong 

>But I don't think creating our own error with the same text will let us catch that one

Ah you are right!

My reason for moving it into a separate type is that we can have a single place where we can update the error message (if at all it changes), instead of multiple places as it is used at a couple of places right now. I will instead create a string const type to be used.